src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -122,47 +146,35 @@ def acr_show(registry_name, resource_group_name=None):
     :param str registry_name: The name of container registry
     :param str resource_group_name: The name of resource group
     """
-    if resource_group_name is None:
-        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
-
+    resource_group_name = get_resource_group_name_by_registry_name(
+        registry_name, resource_group_name)
     client = get_acr_service_client().registries
 
     return client.get(resource_group_name, registry_name)
 
 
-def acr_update_get(client,
-                   registry_name,
-                   resource_group_name=None):
-    if resource_group_name is None:
-        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
-
-    props = client.get(resource_group_name, registry_name)
-
-    return RegistryUpdateParameters(
-        tags=props.tags,
-        admin_user_enabled=props.admin_user_enabled,
-        storage_account=props.storage_account
-    )
+def acr_update_get(client):  # pylint: disable=unused-argument
+    """Returns an empty RegistryUpdateParameters object.
+    """
+    return RegistryUpdateParameters()
 
 
 def acr_update_custom(instance,
-                      admin_enabled=None,
                       storage_account_name=None,
+                      admin_enabled=None,
                       tags=None):
-    if admin_enabled is not None:
-        instance.admin_user_enabled = admin_enabled == ['true']
-
-    if tags is not None:
-        instance.tags = tags
-
     if storage_account_name is not None:
-        storage_account_key = \
-            get_access_key_by_storage_account_name(storage_account_name)
-        storage_account = StorageAccountParameters(
+        storage_account_key = get_access_key_by_storage_account_name(storage_account_name)
+        instance.storage_account = StorageAccountParameters(
             storage_account_name,
             storage_account_key
         )
-    instance.storage_account = storage_account if storage_account_name else None
+
+    if admin_enabled is not None:
```

Comment:
If I understand correctly, we will need a pair of `switch` options support both enable and disable admin user. We used to have it implemented that way (in fact in Powershell cmdlets we still have a pair of switch parameters) but was updated to the current implementation as suggested.